File: Server/models/Feed.js
const Schema = require('mongoose').Schema;
const model = require('mongoose').model;

const feedSchema = Schema({
  message: String,
  date: Date,
  comments: [{
    message: String,
    date: Date,
    sender: {type: Schema.Types.ObjectId, ref: 'User'}
  }],
  starter: {type: Schema.Types.ObjectId, ref: 'User'}
});

const Feed = model('Feed', feedSchema);
module.exports = Feed;

File: DichieApp/app/src/main/java/ds/nutgarden/data/model/Feed.kt
package ds.nutgarden.data.model

import com.google.gson.annotations.SerializedName
import java.util.*

data class Feed(
    @SerializedName("_id") val id: String,
    @SerializedName("message") val message: String,
    @SerializedName("date") val date: Date,
    @SerializedName("comments") val comments: List<Comment>,
    @SerializedName("starter") val starter: User,
    @SerializedName("__v") val version: Int
) {}
